beacon-chain/rpc/prysm/v1alpha1/validator/unblinder.go

```diff
@@ -0,0 +1,201 @@
+	sig := u.b.Signature()
+	blsToExecChanges, err := u.b.Block().Body().BLSToExecutionChanges()
+	if err != nil && !errors.Is(err, consensusblocks.ErrUnsupportedGetter) {
+		return nil, errors.Wrap(err, "could not get bls to execution changes")
+	}
+
+	psb, err := u.blindedProtoBlock()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get blinded proto block")
+	}
+	sb, err := consensusblocks.NewSignedBeaconBlock(psb)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create signed block")
+	}
+	sb.SetSlot(u.b.Block().Slot())
+	sb.SetProposerIndex(u.b.Block().ProposerIndex())
+	sb.SetParentRoot(parentRoot[:])
+	sb.SetStateRoot(stateRoot[:])
+	sb.SetRandaoReveal(randaoReveal[:])
+	sb.SetEth1Data(u.b.Block().Body().Eth1Data())
+	sb.SetGraffiti(graffiti[:])
+	sb.SetProposerSlashings(u.b.Block().Body().ProposerSlashings())
+	sb.SetAttesterSlashings(u.b.Block().Body().AttesterSlashings())
+	sb.SetAttestations(u.b.Block().Body().Attestations())
+	sb.SetDeposits(u.b.Block().Body().Deposits())
+	sb.SetVoluntaryExits(u.b.Block().Body().VoluntaryExits())
+	if err = sb.SetSyncAggregate(agg); err != nil {
+		return nil, errors.Wrap(err, "could not set sync aggregate")
+	}
+	sb.SetSignature(sig[:])
+	if err = sb.SetBLSToExecutionChanges(blsToExecChanges); err != nil && !errors.Is(err, consensusblocks.ErrUnsupportedGetter) {
+		return nil, errors.Wrap(err, "could not set bls to execution changes")
+	}
+	if err = sb.SetExecution(h); err != nil {
+		return nil, errors.Wrap(err, "could not set execution")
+	}
+
+	payload, err := u.builder.SubmitBlindedBlock(ctx, sb)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not submit blinded block")
+	}
+	headerRoot, err := h.HashTreeRoot()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get header root")
+	}
+	payloadRoot, err := payload.HashTreeRoot()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get payload root")
+	}
+	if headerRoot != payloadRoot {
+		return nil, fmt.Errorf("header and payload root do not match, consider disconnect from relay to avoid further issues, "+
+			"%#x != %#x", headerRoot, payloadRoot)
+	}
+
+	agg, err = sb.Block().Body().SyncAggregate()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get sync aggregate")
+	}
+	parentRoot = sb.Block().ParentRoot()
+	stateRoot = sb.Block().StateRoot()
+	randaoReveal = sb.Block().Body().RandaoReveal()
+	graffiti = sb.Block().Body().Graffiti()
+	sig = sb.Signature()
+	blsToExecChanges, err = sb.Block().Body().BLSToExecutionChanges()
+	if err != nil && !errors.Is(err, consensusblocks.ErrUnsupportedGetter) {
+		return nil, errors.Wrap(err, "could not get bls to execution changes")
+	}
+
+	bb, err := u.protoBlock()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get proto block")
+	}
+	wb, err := consensusblocks.NewSignedBeaconBlock(bb)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create signed block")
+	}
+	wb.SetSlot(sb.Block().Slot())
```

Comment:
is there any pattern we can follow so we don't miss/forget doing these sets? also might be worth breaking into its own function to make this more readable.

Reply:
I guess the same comment applies to line 66and onwards. 
do the settings in sb and wb need to be the same? maybe there can be a check on that too somehow?

Reply:
or better yet a utility function where wb/sb are passed so it's at least 1 spot ( not sure if it's completed replicated logic however.

Reply:
I extracted common code